File: OCDemo/OCDemo/设计模式/Dog.m
//
//  Dog.m
//  OCDemo
//
//  Created by mkrq-yh on 2019/3/21.
//  Copyright © 2019 mkrq-yh. All rights reserved.
//

#import "Dog.h"

@implementation Dog

- (void)printClassName {
    NSLog(@"%@---%@",NSStringFromClass([self class]),self);
}

- (void)dogCall {
    NSLog(@"汪汪汪！");
}

@end

File: SwiftDemo/SwiftDemo/设计模式/Dog.swift
//
//  Dog.swift
//  SwiftDemo
//
//  Created by mkrq-yh on 2019/3/21.
//  Copyright © 2019 mkrq-yh. All rights reserved.
//

import UIKit

class Dog: Animal {

    func dogCall() {
        print("汪汪汪！")
    }
    
    override func printClassName() {
        print("\(NSStringFromClass(self.classForCoder))")
    }
}
